.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,34 @@
+name: Tests
+
+on: [push] # without pull_request for the moment
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  specs:
+    name: 'run-specs'
+    # skip on [ci skip] and do not run 2 on push and interal PR
+    if: (contains(github.event.commits[0].message, '[ci skip]') == false) &&  (github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository)
+    continue-on-error: ${{ matrix.allow_failure || false }}
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: ['2.3', '2.4', '2.5', '2.6', '2.7', '3.0']
+
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+
+      - name: Setup & run specs
+        run: |
+          cd gem
```

Comment:
gotta add `set -o pipefail` or a bunch of `&& \` everywhere if you want to fail early

otherwise rspec might fail but the whole run will succeed if `rake install` returns 0